src/AsmResolver.PE.File/PEFile.cs

```diff
@@ -445,6 +447,30 @@ public void AlignSections()
                 header.VirtualSize = section.Contents.GetVirtualSize().Align(OptionalHeader.SectionAlignment);
             }
         }
+
+        public void AlignDataDirectoryEntries(IList<SectionHeader> oldHeaders) {
+            IList<DataDirectory> dataDirectoryEntries = OptionalHeader.DataDirectories;
+            for (int j = 0; j < dataDirectoryEntries.Count; j++)
+            {
+                var DataDirectory = dataDirectoryEntries[j];
+                if (DataDirectory.IsPresentInPE)
+                {
+                    uint VAImageDir = DataDirectory.VirtualAddress;
+                    for(int i = 0; i < oldHeaders.Count; i++)
+                    {
+                        var header = oldHeaders[i];
+                        /* Locate section containing image directory. */
+                        if (header.VirtualAddress <= VAImageDir && header.VirtualAddress + header.SizeOfRawData > VAImageDir)
+                        {
+                            /* Calculate the delta between the new section.rva and the old one */
+                            uint sectionRvaDelta = Sections[i].Rva - header.VirtualAddress;
```

Comment:
What if RVA < VirtualAddress?

Reply:
What about creating an additional GetSectionContainingRva(uint rva, IList<PESection> sections) and the respective TryGetSectionContainingRva? So it would raise ArgumentOutOfRangeException.